cli/src/pcluster/config/cluster_config.py

```diff
@@ -1429,17 +1467,20 @@ def _register_validators(self):
         )
 
         for queue in self.scheduling.queues:
-            self._register_validator(ComputeResourceLaunchTemplateValidator, queue=queue, ami_id=self.ami_id)
+            self._register_validator(
+                ComputeResourceLaunchTemplateValidator, queue=queue, ami_id=self.image_dict[queue.name]
+            )
             for compute_resource in queue.compute_resources:
-                if self.ami_id:
+                if self.image_dict[queue.name]:
                     self._register_validator(
                         InstanceTypeBaseAMICompatibleValidator,
                         instance_type=compute_resource.instance_type,
-                        image=self.ami_id,
+                        image=self.image_dict[queue.name],
                     )
                 self._register_validator(
                     InstanceArchitectureCompatibilityValidator,
                     instance_type=compute_resource.instance_type,
+                    # FIXME: head_node.architecture vs compute_resource.architecture?
```

Comment:
`head_node.architecture` is correct because the validator does't allow to use different architecture